docs/index.asciidoc

```diff
@@ -518,6 +521,17 @@ overwritten with a warning.
 The data stream type used to construct the data stream at index time.
 Currently, only `logs`, `metrics`, `synthetics` and `traces` are supported.
 
+[id="plugins-{type}s-{plugin}-dlq_custom_codes"]
+===== `dlq_custom_codes`
+
+* Value type is <<number,number>>
+* Default value is `[]`.
+
+List HTTP's error codes that are considered valid to move the events into the dead letter queue.
+This list is an addition to the ordinary error codes considered for this feature, 400 and 404.
+It's considered a configuration error to re-use the same predefined codes for success, DLQ or conflict.
```

Comment:
It looks like we already handle `200`/`201` as explicit success, `400`/`404` as DLQ, and `409` as conflict, and are declaring it a configuration error to re-use any of these.

If a user can't use this to intercept `409` conflicts, or 404's attempting to update records that don't exist, what error codes do we expect? Only authorization-related ones like `403`? Others? Do we have real-world use-cases driving this feature?

Reply:
This PR aim to solve request #697 which explicitly state that `403` error code could be eventually added as a DLQ trigger code.
While intersection with existing DLQ codes (`400` and `404`) could be seen as not necessarily a conflict, because both trigger the same feature, using error codes already configured as success (`200`/`201`) document conflict error (`409`) generates confusion in the precedence order when the [code process responses](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/1cab1ad2a41b28ac569cb77809b48ed6b66d2b39/lib/logstash/plugin_mixins/elasticsearch/common.rb#L264-L280).

Suppose that also `409` is customized as a `dlq_custom_code` what should take precedence? Processing it as a failed action or as DLQ event?

To keep the things simple I imagined to keep out of the set the codes that already has a meaning in the plugin.

Reply:
@yaauie seems that this needs your feedback?

Reply:
Yes. I can unblock this PR.

We can consider a later feature to enable this explicit custom-code routing to also intercept `400`/`404`/`409` that are already routed by the implementation, and if and when we do so we have a path to do it in a non-breaking way.